### PR TITLE
Fix for FeatureLayers w/ Polygons

### PR DESCRIPTION
**Pull request by bhaskarvk**
This is a rework of #101 but also supports searching added Shapes (L.Path/L.MultiPolyline/L.MultiPolygon )in addition to L.Marker/L.CircleMarker. 
Secondly it also supports search Shapes in GeoJSONs in addition to Markers.

### CODE REVIEW

**Comment by bhaskarvk** [comment]
Hi,
 Any interest in merging this PR?

**Comment by stefanocudini** [comment]
hi @bhaskarvk I'm currently evaluating your PR

**Comment by bhaskarvk** [comment]
Thanks @stefanocudini. Let me know if you need any clarification.

**Comment by stefanocudini** [comment]
No @bhaskarvk it's ok!
Large contributions require large checks! ;-)

In the last time I have all my free time engaged in the development and promotion of another opensource project [KeplerJs](https://github.com/keplerjs)
You probably may be interested it... I suggest you give it a look!

Cheers
Sefano

**Comment by bhaskarvk** [comment]
Thanks @stefanocudini KeplerJs looks very interesting. Thanks for the pointer.

**Comment by stefanocudini** [comment]
hi @bhaskarvk  good contribution!

I just merged and published!
If you were using the plugin in some project you can include it with the NPM version **2.3.0**

**Comment by bhaskarvk** [comment]
Awesome and thanks! I don't use it in my `leaflet.extras` package which is an R wrapper for many leaflet plugins including this one.

File: src/leaflet-search.js [hunk 1]
```diff
@@ -519,79 +519,82 @@ L.Control.Search = L.Control.extend({
...
+        loc = layer.getLatLng();
+        loc.layer = layer;
+        retRecords[ that._getPath(layer.feature.properties,propName) ] = loc;
+      }
+      else {
+        //throw new Error("propertyName '"+propName+"' not found in marker"); 
+        console.warn("propertyName '"+propName+"' not found in marker"); 
+      }
+    }
+    if(layer instanceof L.Path || layer instanceof L.MultiPolyline || layer instanceof L.MultiPolygon)
```
Comment (stefanocudini) [inline src/leaflet-search.js]: @bhaskarvk what's L.MultiPolyline?!???  Not exists this in Leaflet

**Pull request merged**